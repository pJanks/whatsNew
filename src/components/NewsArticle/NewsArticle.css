img {
  height: 350px;
  width: 300px;
}

p,
h2{
  color: black;
  margin: 10px;
  width: 300%;
}

article {
  display: flex;
  flex-direction: column;
  height: 400px;
  margin: 30px;
  margin-bottom: 200px;
  text-align: center;
  width: 30%;
}
