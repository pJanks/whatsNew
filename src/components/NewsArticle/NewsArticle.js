import React from 'react';
import './NewsArticle.css';

const NewsArticle = (props) => {
  return (<article key={props.article.id}>
    <h2>{props.article.headline}</h2>
    <img src={props.article.img}/>
    <p>{props.article.description}</p>
  </article>)
}

export default NewsArticle;
