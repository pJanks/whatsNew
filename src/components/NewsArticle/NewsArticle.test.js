import React from 'react';
import ReactDOM from 'react-dom';
import NewsArticle from './NewsArticle';
import { shallow } from 'enzyme';

describe('NewsArticle', () => {
  let wrapper, instance, mockNewsArticle;

  beforeEach(() => {
    mockNewsArticle = {
      id: 3,
      headline: 'crazy news',
      img: 'aPicture',
      description: 'something crazy',
      url: 'www.hi.com'
    }
    wrapper = shallow(<NewsArticle key={mockNewsArticle.id} article={mockNewsArticle} />)
    instance = wrapper.instance()
  });

  it('should match the snapshot', () => {

    expect(wrapper).toMatchSnapshot()
  });
});
