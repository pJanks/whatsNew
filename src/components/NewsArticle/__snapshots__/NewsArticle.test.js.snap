// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsArticle should match the snapshot 1`] = `
<article
  key="3"
>
  <h2>
    crazy news
  </h2>
  <img
    src="aPicture"
  />
  <p>
    something crazy
  </p>
  <a
    href="www.hi.com"
  >
    Click for full article
  </a>
</article>
`;
